src/p5.dance.js

```diff
@@ -134,6 +138,68 @@ module.exports = class DanceParty {
     this.p5_.remove();
   }
 
+  loadSprites(animationData, spriteNames) {
+    const promises = [];
+    spriteNames.forEach(costume => {
+      if (this.animations[costume].length === this.world.MOVE_NAMES.length) {
+        // Already loaded, nothing to do:
+        return;
+      }
+      // We have found a new sprite to load, reset allSpritesLoaded to false
+      this.allSpritesLoaded = false;
+      const costumeData = animationData[costume.toLowerCase()];
+      this.world.MOVE_NAMES.forEach(({ name: moveName, mirror }, moveIndex) => {
+        const moveData = costumeData[moveName.toLowerCase()];
+
+        promises.push(new Promise(resolveSpriteSheet => {
```

Comment:
Cleanup idea: ResourceLoader's `loadSpriteSheet` can return a promise instead of taking a callback.

Reply:
Sure, that would be a nice enhancement. As you know, this was written before ResourceLoader, so we were forced to adapt p5’s callbacks to promises. Now we have a layer in between.

Reply:
Updated with this change. 👍